core/types/gen_header_json.go

```diff
@@ -12,24 +12,25 @@ import (
 )
 
 func (h Header) MarshalJSON() ([]byte, error) {
-	type HeaderJSON struct {
-		ParentHash  common.Hash    `json:"parentHash"`
-		UncleHash   common.Hash    `json:"sha3Uncles"`
-		Coinbase    common.Address `json:"miner"`
-		Root        common.Hash    `json:"stateRoot"`
-		TxHash      common.Hash    `json:"transactionsRoot"`
-		ReceiptHash common.Hash    `json:"receiptsRoot"`
-		Bloom       Bloom          `json:"logsBloom"`
-		Difficulty  *hexutil.Big   `json:"difficulty"`
-		Number      *hexutil.Big   `json:"number"`
-		GasLimit    *hexutil.Big   `json:"gasLimit"`
-		GasUsed     *hexutil.Big   `json:"gasUsed"`
-		Time        *hexutil.Big   `json:"timestamp"`
-		Extra       hexutil.Bytes  `json:"extraData"`
-		MixDigest   common.Hash    `json:"mixHash"`
-		Nonce       BlockNonce     `json:"nonce"`
-	}
-	var enc HeaderJSON
+	type Header struct {
+		ParentHash  common.Hash    `json:"parentHash"       gencodec:"required"`
+		UncleHash   common.Hash    `json:"sha3Uncles"       gencodec:"required"`
+		Coinbase    common.Address `json:"miner"            gencodec:"required"`
+		Root        common.Hash    `json:"stateRoot"        gencodec:"required"`
+		TxHash      common.Hash    `json:"transactionsRoot" gencodec:"required"`
+		ReceiptHash common.Hash    `json:"receiptsRoot"     gencodec:"required"`
+		Bloom       Bloom          `json:"logsBloom"        gencodec:"required"`
+		Difficulty  *hexutil.Big   `json:"difficulty"       gencodec:"required"`
+		Number      *hexutil.Big   `json:"number"           gencodec:"required"`
+		GasLimit    *hexutil.Big   `json:"gasLimit"         gencodec:"required"`
+		GasUsed     *hexutil.Big   `json:"gasUsed"          gencodec:"required"`
+		Time        *hexutil.Big   `json:"timestamp"        gencodec:"required"`
+		Extra       hexutil.Bytes  `json:"extraData"        gencodec:"required"`
+		MixDigest   common.Hash    `json:"mixHash"          gencodec:"required"`
+		Nonce       BlockNonce     `json:"nonce"            gencodec:"required"`
```

Comment:
Is there any reason why all the `gencodec:"required"` tags are preserved within the generated code? As I understand it correctly, these fields are used to signal to the unmarshal that a missing field should cause an error. However that is "hard coded" in the generated unmarshal method, so I don't see why we'd need to preserve the tags in the generated code too.